processor/metricstransformprocessor/metrics_transform_processor_helper_test.go

```diff
@@ -0,0 +1,671 @@
+								Action:          AggregateLabelValues,
+								NewValue:        "new/label2-value",
+								AggregationType: Sum,
+							},
+							labelSetMap:         map[string]bool{"label2": true},
+							aggregatedValuesSet: map[string]bool{"label2-value1": true, "label2-value2": true},
+						},
+					},
+				},
+			},
+			in: []*metricspb.Metric{
+				testcaseBuilder().setName("metric1").setLabels([]string{"label1", "label2"}).setDataType(metricspb.MetricDescriptor_GAUGE_INT64).
+					addTimeseries(2, []string{"label1-value1", "label2-value1"}).addTimeseries(1, []string{"label1-value1", "label2-value2"}).
+					addInt64Point(0, 3, 2).addInt64Point(1, 1, 2).
+					build(),
+			},
+			out: []*metricspb.Metric{
+				testcaseBuilder().setName("metric1").setLabels([]string{"label1", "label2"}).setDataType(metricspb.MetricDescriptor_GAUGE_INT64).
+					addTimeseries(2, []string{"label1-value1", "label2-value1"}).addTimeseries(1, []string{"label1-value1", "label2-value2"}).
+					addInt64Point(0, 3, 2).addInt64Point(1, 1, 2).
+					build(),
+				testcaseBuilder().setName("metric1").setLabels([]string{"label1", "label2"}).setDataType(metricspb.MetricDescriptor_GAUGE_INT64).
+					addTimeseries(1, []string{"label1-value1", "new/label2-value"}).
+					addInt64Point(0, 4, 2).
+					build(),
+			},
+		},
+		{
+			name: "metric_labels_aggregation_sum_distribution_insert",
+			transforms: []mtpTransform{
+				{
+					MetricName: "metric1",
+					Action:     Insert,
+					Operations: []mtpOperation{
+						{
+							configOperation: Operation{
+								Action:          AggregateLabels,
+								AggregationType: Sum,
+							},
+							labelSetMap: map[string]bool{"label1": true},
+						},
+					},
+				},
+			},
+			in: []*metricspb.Metric{
+				testcaseBuilder().setName("metric1").setLabels([]string{"label1", "label2"}).setDataType(metricspb.MetricDescriptor_GAUGE_DISTRIBUTION).
+					addTimeseries(1, []string{"label1-value1", "label2-value1"}).addTimeseries(3, []string{"label1-value1", "label2-value2"}).
+					addDistributionPoints(0, 1, 3, 6, []float64{1, 2}, []int64{0, 1, 2}, 3).
+					addDistributionPoints(1, 1, 5, 10, []float64{1, 2}, []int64{1, 1, 3}, 4).
+					build(),
+			},
+			out: []*metricspb.Metric{
+				testcaseBuilder().setName("metric1").setLabels([]string{"label1", "label2"}).setDataType(metricspb.MetricDescriptor_GAUGE_DISTRIBUTION).
+					addTimeseries(1, []string{"label1-value1", "label2-value1"}).addTimeseries(3, []string{"label1-value1", "label2-value2"}).
+					addDistributionPoints(0, 1, 3, 6, []float64{1, 2}, []int64{0, 1, 2}, 3).
+					addDistributionPoints(1, 1, 5, 10, []float64{1, 2}, []int64{1, 1, 3}, 4).
+					build(),
+				testcaseBuilder().setName("metric1").setLabels([]string{"label1"}).setDataType(metricspb.MetricDescriptor_GAUGE_DISTRIBUTION).
+					addTimeseries(1, []string{"label1-value1"}).
+					addDistributionPoints(0, 1, 8, 16, []float64{1, 2}, []int64{1, 2, 5}, 7).
+					build(),
+			},
+		},
+	}
+)
```

Comment:
- Pls make sure you have a tests that has 2+ distribution points
- Pls make sure you have a tests that has 3+ distribution buckets

(you may already, I didn't check that thoroughly)